### DISCUSSION TITLE
Retrieve flags value in custom route

### DISCUSSION THREAD

**Discussion by DblK** [pocketbase/pocketbase#3167] [Q&A] [answered]
Hi,

I trying to test out pocketbase to ease some backend work and I need to have a way to retrieve the address the serve flag will use to use it in another part of the app.
How this could be done ?

Is there a way to use viper config with pocketbase to load the config from a file before running?

Thx for the help

**Comment by ganigeorgiev** [comment, marked as answer]
>  ... I need to have a way to retrieve the address the serve flag will use to use it in another part of the app. How this could be done ?

The server address is not part of the app interface so it is not stored there but you can read the flags from `os.Args` and/or explore the fields of the cobra instance under `app.RootCmd` and its `serve` child command. But keep in mind that we also apply defaults in case the flags are missing so it really would depend from where and at what stage you want to access the server address(es).

One possible approach could be to read the address right before initializing the server and store it in some custom variable by attaching to the `OnBeforeServe` hook:

```go
app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
	log.Println(e.Server.Addr)

	return nil
})
```


> Is there a way to use viper config with pocketbase to load the config from a file before running?

Yes, check [`pocketbase.NewWithConfig()`](https://pkg.go.dev/github.com/pocketbase/pocketbase#NewWithConfig) (_it is up to the developers to decide from where and how the `pocketbase.Config` struct will be loaded_).

**Reply by ganigeorgiev** [reply]
_I forgot to mention that if you want to initialize the web server programmatically you ca also call [`apis.Serve(app, config)`](https://pkg.go.dev/github.com/pocketbase/pocketbase/apis#Serve)._
